QUARKUS-4592.md

```diff
@@ -0,0 +1,112 @@
+From what I can see, the old ways to configure TLS in extensions are still valid and for now, the TLS registry is added as additional option.
+User-wise, apart from having to configure TLS just once, the biggest change is periodic certificate reloading, 
+Let's Encrypt support and OpenShift documentation explaining how to configure the Deployment.
+
+What has changed in Quarkus extensions so far (remember this is living organism):
+- OTel OTLP span exporter allows to communicate with an OTLP collector via secured channel
+- OIDC global `trust all` flag can be configured from default TLS registry
+- SmallRye Reactive Messaging allows to configure TLS with the registry in addition to the old ways
+- REDIS client can be configured with the registry (ALPN, hostname verification algorithm, trust all flag, PEM/JKS/PFX and client certs)
+- Quarkus REST client has configuration property for named configuration bucket in addition to the old ways
+- AMQ / MQTT / Pulsar and RabbitMQ are also configurable from the registry
+- WebSockets NEXT integration is also finished (configures underlying HTTP client options under Vert.x WS)
+- gRPC extension also propagates TLS configuration from the registry to underlying Vert.x client for the gRPC clients
+
+## Scope of the testing
+Main focus is on testing features documented in the TLS registry reference. That is:
+- configure HTTP server to use HTTPS protocol
+- configure management interface to use HTTPS protocol
+- configure mTLS
+- configure client to use HTTPS protocol; here we should concentrate on:
+  - gRPC
+  - Quarkus REST client
+- accessing TLS registry API from CDI in application code (both named and the default one)
+- configure keystore, truststore, cipher suits and protocol
+- use PEM / PKCS12 / JKS keystore and truststore
+- test certificate reloading in DEV mode
+- use OpenShift serving certificates to generate and renew TLS certificates automatically
+- generate certificates and DEV CA with Quarkus CLI, then run app and make sure the certs are configured and valid
+- support for the TLS registry to the mailer extension
+
+Additionally, secured communication between Quarkus application and a Strimzi server must be tested.
+We want to test registry where we already test extension-specific TLS configuration, while test for every single supported extension may not be possible due to time limitations.
+For example, secured communication with AMQ broker, REDIS cache and OTLP collector will be tested if time allows, or delayed (with opened Quarkus QE Test Suite issue).
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Focus on exploratory testing of the feature
+- Ensure good user experience and simplicity of use
+
+## Existing test coverage
+Quarkus unit test coverage is sufficient. As much as I am thinking how much do we need to test.
+Integration tests are spare and sometimes missing, for example I didn't find Quarkus REST client or OTLP integration 
```

Comment:
I plan to add it to TS. Upstream doesn't test every feature in native. It would not be bearable for upstream CI. At least this way, we will have something upstream doesn't.

Reply:
And now I have found that later Clement added dedicated PR for ITs. I'll drop part of this sentence from TP.

Reply:
https://github.com/quarkusio/quarkus/pull/42042/files

Reply:
Dropped, so closing this comment thread.